tests/datasets/test_l7irish.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import glob
+import os
+import shutil
+from datetime import datetime
+from pathlib import Path
+
+import pytest
+import matplotlib.pyplot as plt
+import torch
+import torch.nn as nn
+# from torch.utils.data import DataLoader
+from _pytest.monkeypatch import MonkeyPatch
+from rasterio.crs import CRS
+
+import torchgeo.datasets.utils
+from torchgeo.samplers import RandomGeoSampler
+from torchgeo.datasets import L7Irish, BoundingBox, IntersectionDataset, UnionDataset
+
+
+def download_url(url: str, root: str, *args: str, **kwargs: str) -> None:
+    shutil.copy(url, root)
+
+# ds = l7irish('data/l7irish', res=300)
+# sampler = RandomGeoSampler(ds, size=5000)
+# dl = DataLoader(ds, sampler=sampler)
+# bbox = BoundingBox(minx, maxx, miny, maxy, mint, maxt)
+# sample = ds[bbox]
+# ds.plot(sample)
+# plt.show()
+
+class TestL7Irish:
+    @pytest.fixture
+    def dataset(self, monkeypatch: MonkeyPatch, tmp_path: Path) -> L7Irish:
+        monkeypatch.setattr(torchgeo.datasets.l7irish, "download_url", download_url)
+        # need to create new md5s?
+        md5s = {
+            "austral": "dbb6b5628f50861b9b89f548d25a925f",
+            "boreal": "cecc72de09aacde4c4f8d7f0cf0d3f6f",
+            "mid_latitude_north": "0f8382ca6554fb7cf9aff42226a14f9d",
+            "mid_latitude_south": "b17cf6d023f752c533211fdb742f296b",
+            "polar_north": "73923dcaf1b9b79bad82de1aa0740d1e",
+            "polar_south": "3bc9f4c6f8955b10b4d55d23e0ab2da7",
+            "subtropical_north": "f8f039970256902e6e9ebd6747589294",
+            "subtropical_south": "8346d73a983396c5d41b577c3a94bc26",
+            "tropical": "abe19b22b5d031e6b609cc7207706c3d",
+        }
+        
+        monkeypatch.setattr(L7Irish, "md5s", md5s)
+        url = os.path.join("tests", "data", "l7irish", "{}.tar.gz")
+        monkeypatch.setattr(L7Irish, "url", url)
+        monkeypatch.setattr(plt, "show", lambda *args: None)
+        root = str(tmp_path)
+        transforms = nn.Identity() # why does it need nn?
```

Comment:
We need some transform to get 100% coverage. It just needs to be an identity function, doesn't need to be PyTorch's.